core-libs/setup/ssr/logger/get-logger-inspect-options.ts

```diff
@@ -0,0 +1,54 @@
+// PRIVATE API
+
+import { isDevMode } from '@angular/core';
+import { InspectOptions } from 'node:util';
+
+/**
+ * Default options to be used for formatting log messages in NodeJS (in both prod and dev mode).
+ *
+ * They are meant to be passed to `inspect()` or `formatWithOptions()` functions from the `node:util` module
+ * for logging purposes.
+ */
+const DEFAULT_LOGGER_INSPECT_OPTIONS: InspectOptions = {
+  /**
+   * Prevent the depth of the logged object to be limited to 2 (which is a NodeJS default).
+   * Otherwise, the object's properties at nested levels higher than 2 would not be visible in the logs.
+   * e.g. `{ a: { b: { c: { d: 'e' } } } }` would be presented as `{ a: { b: { c: [Object] } } }` in the logs.
+   *
+   * The value 10 was chosen arbitrarily. It can be adjusted in the future, if needed.
+   */
+  depth: 10,
```

Comment:
`query:` Shouldn't we consider giving a lower default value? I wonder from the angle of the amount of logs and the app performance.

Reply:
TBH I had hard time deciding on the arbitrary value.
- 2 (default) seemed too low. especially for `logObject.context.error.cause.headers` it caused `headers` to be displayed as `Map`
- null/Infinity seemed unnecessarily risky in prod mode, if someone accidentally did log a super big object
- When reading logs, especially trying to understand a reason of a system failure, what matters is having the full picture of what happened. It would be a pity, if important details are abbreviated
That's why I've guessed the number 10, which is generous enough (subjectively IMHO), and also prevents too deep inspection levels.

Reply:
Thank you for your explanation. I do agree with the reasoning.
In that case, maybe we should consider making this API public So customers can use it in their custom loggers and override the `depth` value if needed.